src/ModelObjects/representations.rs

```diff
@@ -143,25 +139,26 @@ impl<'a> SystemRepresentation {
         action: &str,
         open_edges: &mut Vec<(&'a Component, Vec<&'a Edge>, usize)>,
         sync_type: &SyncType,
-    ) -> bool {
+    ) {
         match self {
             SystemRepresentation::Composition(left_side, right_side) => {
-                left_side.collect_open_edges(states, index, action, open_edges, sync_type)
-                    || right_side.collect_open_edges(states, index, action, open_edges, sync_type)
+                left_side.collect_open_edges(states, index, action, open_edges, sync_type);
+                right_side.collect_open_edges(states, index, action, open_edges, sync_type);
             }
             SystemRepresentation::Conjunction(left_side, right_side) => {
-                let open_edges_len = open_edges.len();
-                if left_side.collect_open_edges(states, index, action, open_edges, sync_type) {
-                    let left_found_transitions = open_edges_len != open_edges.len();
-                    if right_side.collect_open_edges(states, index, action, open_edges, sync_type) {
-                        let right_found_transitions = open_edges_len != open_edges.len();
-                        return left_found_transitions == right_found_transitions;
-                    }
+                let mut left = vec![];
+                let mut right = vec![];
+
+                left_side.collect_open_edges(states, index, action, &mut left, sync_type);
+                right_side.collect_open_edges(states, index, action, &mut right, sync_type);
+
+                if !(left.is_empty() || right.is_empty()) {
+                    open_edges.append(&mut left);
+                    open_edges.append(&mut right);
```

Comment:
This is the only the only change to functionality so far, everything else is refactoring allowed by these changes.

The idea of the fix: Given the transition system T = A x B where A and B are automatons in conjunction.
The old code would fail if A had an open transition and B did not have a matching transition.
I argue that no errors should be thrown in this case and we instead simply ignore that case.

Now the code only adds an edge to T when both A and B have matching transitions as per equation 7
![image](https://user-images.githubusercontent.com/10059450/122677073-f3e9d700-d1e0-11eb-9fcd-4aa57fa61b66.png)

I might have errors, as my understanding of how inconsistency in conjunctions should be handled is severely limited.
Additionally I don't know how delays `a in R>=0` is handled.